// @primer/react/lib-esm/PageLayout/PageLayout.stories.js
import React from 'react';
import { Box, BranchName, Button, Heading, Link, StateLabel, TabNav, Text } from '..';
import { NavList } from '../NavList';
import { Placeholder } from '../Placeholder';
import { PageLayout } from './PageLayout';
const meta = {
  title: 'Layout/PageLayout',
  component: PageLayout,
  parameters: {
    layout: 'fullscreen',
    controls: {
      expanded: true
    }
  }
};
export const Playground = args => /*#__PURE__*/React.createElement(PageLayout, {
  containerWidth: args.containerWidth,
  padding: args.padding,
  rowGap: args.rowGap,
  columnGap: args.columnGap,
  sx: args.sx
}, args['Show header?'] ? /*#__PURE__*/React.createElement(PageLayout.Header, {
  divider: args['Header.divider'],
  dividerWhenNarrow: args['Header.dividerWhenNarrow']
}, /*#__PURE__*/React.createElement(Placeholder, {
  height: 64,
  label: "Header"
})) : null, /*#__PURE__*/React.createElement(PageLayout.Content, {
  width: args['Content.width']
}, /*#__PURE__*/React.createElement(Placeholder, {
  height: 400,
  label: "Content"
})), args['Show pane?'] ? /*#__PURE__*/React.createElement(PageLayout.Pane, {
  position: args['Pane.position'],
  positionWhenNarrow: args['Pane.positionWhenNarrow'],
  width: args['Pane.width'],
  divider: args['Pane.divider'],
  dividerWhenNarrow: args['Pane.dividerWhenNarrow']
}, /*#__PURE__*/React.createElement(Placeholder, {
  height: 200,
  label: "Pane"
})) : null, args['Show footer?'] ? /*#__PURE__*/React.createElement(PageLayout.Footer, {
  divider: args['Footer.divider'],
  dividerWhenNarrow: args['Footer.dividerWhenNarrow']
}, /*#__PURE__*/React.createElement(Placeholder, {
  height: 64,
  label: "Footer"
})) : null);
Playground.displayName = "Playground";
Playground.argTypes = {
  'Show header?': {
    type: 'boolean',
    defaultValue: true,
    table: {
      category: 'Header'
    }
  },
  'Header.divider': {
    type: {
      name: 'enum',
      value: ['none', 'line']
    },
    defaultValue: 'none',
    control: {
      type: 'radio'
    },
    table: {
      category: 'Header',
      defaultValue: {
        summary: '"none"'
      }
    }
  },
  'Header.dividerWhenNarrow': {
    type: {
      name: 'enum',
      value: ['inherit', 'none', 'line', 'filled']
    },
    defaultValue: 'inherit',
    control: {
      type: 'radio'
    },
    table: {
      category: 'Header',
      defaultValue: {
        summary: '"inherit"'
      }
    }
  },
  'Content.width': {
    type: {
      name: 'enum',
      value: ['full', 'medium', 'large', 'xlarge']
    },
    defaultValue: 'full',
    control: {
      type: 'radio'
    },
    table: {
      category: 'Content',
      defaultValue: {
        summary: '"full"'
      }
    }
  },
  'Show pane?': {
    type: 'boolean',
    defaultValue: true,
    table: {
      category: 'Pane'
    }
  },
  'Pane.position': {
    type: {
      name: 'enum',
      value: ['start', 'end']
    },
    defaultValue: 'end',
    control: {
      type: 'radio'
    },
    table: {
      category: 'Pane',
      defaultValue: {
        summary: '"end"'
      }
    }
  },
  'Pane.positionWhenNarrow': {
    type: {
      name: 'enum',
      value: ['inherit', 'start', 'end']
    },
    defaultValue: 'inherit',
    control: {
      type: 'radio'
    },
    table: {
      category: 'Pane',
      defaultValue: {
        summary: '"inherit"'
      }
    }
  },
  'Pane.width': {
    type: {
      name: 'enum',
      value: ['small', 'medium', 'large']
    },
    defaultValue: 'medium',
    control: {
      type: 'radio'
    },
    table: {
      category: 'Pane',
      defaultValue: {
        summary: '"medium"'
      }
    }
  },
  'Pane.divider': {
    type: {
      name: 'enum',
      value: ['none', 'line']
    },
    defaultValue: 'none',
    control: {
      type: 'radio'
    },
    table: {
      category: 'Pane',
      defaultValue: {
        summary: '"none"'
      }
    }
  },
  'Pane.dividerWhenNarrow': {
    type: {
      name: 'enum',
      value: ['inherit', 'none', 'line', 'filled']
    },
    defaultValue: 'inherit',
    control: {
      type: 'radio'
    },
    table: {
      category: 'Pane',
      defaultValue: {
        summary: '"inherit"'
      }
    }
  },
  'Show footer?': {
    type: 'boolean',
    defaultValue: true,
    table: {
      category: 'Footer'
    }
  },
  'Footer.divider': {
    type: {
      name: 'enum',
      value: ['none', 'line']
    },
    defaultValue: 'none',
    control: {
      type: 'radio'
    },
    table: {
      category: 'Footer',
      defaultValue: {
        summary: '"none"'
      }
    }
  },
  'Footer.dividerWhenNarrow': {
    type: {
      name: 'enum',
      value: ['inherit', 'none', 'line', 'filled']
    },
    defaultValue: 'inherit',
    control: {
      type: 'radio'
    },
    table: {
      category: 'Footer',
      defaultValue: {
        summary: '"inherit"'
      }
    }
  }
};
Playground.args = {
  containerWidth: 'xlarge',
  padding: 'normal',
  rowGap: 'normal',
  columnGap: 'normal'
};
export const PullRequestPage = () => /*#__PURE__*/React.createElement(PageLayout, null, /*#__PURE__*/React.createElement(PageLayout.Header, null, /*#__PURE__*/React.createElement(Box, {
  sx: {
    display: 'flex',
    flexDirection: 'column',
    gap: 3
  }
}, /*#__PURE__*/React.createElement(Box, null, /*#__PURE__*/React.createElement(Heading, {
  as: "h1",
  sx: {
    fontWeight: 'normal'
  }
}, "Input validation styles ", /*#__PURE__*/React.createElement(Text, {
  sx: {
    color: 'fg.muted',
    fontWeight: 'light'
  }
}, "#1831")), /*#__PURE__*/React.createElement(Box, {
  sx: {
    display: 'flex',
    gap: 2,
    alignItems: 'center'
  }
}, /*#__PURE__*/React.createElement(StateLabel, {
  status: "pullOpened"
}, "Open"), /*#__PURE__*/React.createElement(Text, {
  sx: {
    fontSize: 1,
    color: 'fg.muted'
  }
}, /*#__PURE__*/React.createElement(Link, {
  href: "#",
  muted: true,
  sx: {
    fontWeight: 'bold'
  }
}, "mperrotti"), ' ', "wants to merge 3 commits into ", /*#__PURE__*/React.createElement(BranchName, {
  href: "#"
}, "main"), " from", ' ', /*#__PURE__*/React.createElement(BranchName, {
  href: "#"
}, "mp/validation-styles")))), /*#__PURE__*/React.createElement(TabNav, null, /*#__PURE__*/React.createElement(TabNav.Link, {
  href: "#",
  selected: true
}, "Conversation"), /*#__PURE__*/React.createElement(TabNav.Link, {
  href: "#"
}, "Commits"), /*#__PURE__*/React.createElement(TabNav.Link, {
  href: "#"
}, "Checks"), /*#__PURE__*/React.createElement(TabNav.Link, {
  href: "#"
}, "Files changed")))), /*#__PURE__*/React.createElement(PageLayout.Content, null, /*#__PURE__*/React.createElement(Box, {
  sx: {
    border: '1px solid',
    borderRadius: 2,
    borderColor: 'border.default',
    height: 200
  }
}), /*#__PURE__*/React.createElement(Box, {
  sx: {
    maxWidth: '100%',
    overflowX: 'auto',
    border: '1px solid',
    whiteSpace: 'nowrap',
    borderColor: 'border.default',
    mt: 3,
    p: 3,
    borderRadius: 2
  }
}, "This box has really long content. If it is too long, it will cause x overflow and should show a scrollbar. When this overflows, it should not break to overall page layout!")), /*#__PURE__*/React.createElement(PageLayout.Pane, null, /*#__PURE__*/React.createElement(Box, {
  sx: {
    display: 'flex',
    flexDirection: 'column',
    gap: 3
  }
}, /*#__PURE__*/React.createElement(Box, null, /*#__PURE__*/React.createElement(Text, {
  sx: {
    fontSize: 0,
    fontWeight: 'bold',
    display: 'block',
    color: 'fg.muted'
  }
}, "Assignees"), /*#__PURE__*/React.createElement(Text, {
  sx: {
    fontSize: 0,
    color: 'fg.muted',
    lineHeight: 'condensed'
  }
}, "No one \u2013", ' ', /*#__PURE__*/React.createElement(Link, {
  href: "#",
  muted: true
}, "assign yourself"))), /*#__PURE__*/React.createElement(Box, {
  role: "separator",
  sx: {
    width: '100%',
    height: 1,
    backgroundColor: 'border.default'
  }
}), /*#__PURE__*/React.createElement(Box, null, /*#__PURE__*/React.createElement(Text, {
  sx: {
    fontSize: 0,
    fontWeight: 'bold',
    display: 'block',
    color: 'fg.muted'
  }
}, "Labels"), /*#__PURE__*/React.createElement(Text, {
  sx: {
    fontSize: 0,
    color: 'fg.muted',
    lineHeight: 'condensed'
  }
}, "None yet")))));
PullRequestPage.displayName = "PullRequestPage";
export const SettingsPage = () => /*#__PURE__*/React.createElement(PageLayout, null, /*#__PURE__*/React.createElement(PageLayout.Pane, {
  position: "start"
}, /*#__PURE__*/React.createElement(NavList, {
  "aria-label": "main"
}, /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#"
}, "Profile"), /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#",
  "aria-current": "page"
}, "Account"), /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#"
}, "Emails"), /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#"
}, "Notifications"))), /*#__PURE__*/React.createElement(PageLayout.Content, null, /*#__PURE__*/React.createElement(Heading, {
  as: "h2",
  sx: {
    fontSize: 4,
    fontWeight: 'normal',
    color: 'danger.fg',
    mb: 2
  }
}, "Danger zone"), /*#__PURE__*/React.createElement(Box, {
  sx: {
    border: '1px solid',
    borderColor: 'danger.emphasis',
    borderRadius: 2,
    p: 3,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    gap: 3
  }
}, /*#__PURE__*/React.createElement(Box, {
  sx: {
    display: 'grid',
    gap: 1
  }
}, /*#__PURE__*/React.createElement(Text, {
  sx: {
    fontSize: 1,
    fontWeight: 'bold',
    color: 'danger.fg'
  }
}, "Delete account"), /*#__PURE__*/React.createElement(Text, {
  sx: {
    fontSize: 1,
    color: 'fg.muted'
  }
}, "Are you sure you don't want to just downgrade your account to a free account? We won't charge your credit card anymore.")), /*#__PURE__*/React.createElement(Button, {
  variant: "danger"
}, "Delete account"))));
SettingsPage.displayName = "SettingsPage";
// TODO: discussions page example
export default meta;
