// @primer/react/lib/FormControl/_FormControlLeadingVisual.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("..");

var _constants = require("../constants");

var _slots = require("./slots");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const FormControlLeadingVisual = ({
  children,
  sx
}) => /*#__PURE__*/_react.default.createElement(_slots.Slot, {
  name: "LeadingVisual"
}, ({
  disabled,
  captionId
}) => /*#__PURE__*/_react.default.createElement(_.Box, {
  color: disabled ? 'fg.muted' : 'fg.default',
  sx: {
    '> *': {
      minWidth: captionId ? (0, _constants.get)('fontSizes.4') : (0, _constants.get)('fontSizes.2'),
      minHeight: captionId ? (0, _constants.get)('fontSizes.4') : (0, _constants.get)('fontSizes.2'),
      fill: 'currentColor'
    },
    ...sx
  },
  ml: 2
}, children));

FormControlLeadingVisual.displayName = "FormControlLeadingVisual";
var _default = FormControlLeadingVisual;
exports.default = _default;
