// @primer/react/lib/deprecated/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BorderBox", {
  enumerable: true,
  get: function () {
    return _BorderBox.default;
  }
});
Object.defineProperty(exports, "ChoiceFieldset", {
  enumerable: true,
  get: function () {
    return _ChoiceFieldset.default;
  }
});
Object.defineProperty(exports, "Item", {
  enumerable: true,
  get: function () {
    return _ChoiceFieldset.Item;
  }
});
Object.defineProperty(exports, "ChoiceInputField", {
  enumerable: true,
  get: function () {
    return _ChoiceInputField.default;
  }
});
Object.defineProperty(exports, "Flex", {
  enumerable: true,
  get: function () {
    return _Flex.default;
  }
});
Object.defineProperty(exports, "Grid", {
  enumerable: true,
  get: function () {
    return _Grid.default;
  }
});
Object.defineProperty(exports, "Position", {
  enumerable: true,
  get: function () {
    return _Position.default;
  }
});
Object.defineProperty(exports, "Absolute", {
  enumerable: true,
  get: function () {
    return _Position.Absolute;
  }
});
Object.defineProperty(exports, "Fixed", {
  enumerable: true,
  get: function () {
    return _Position.Fixed;
  }
});
Object.defineProperty(exports, "Relative", {
  enumerable: true,
  get: function () {
    return _Position.Relative;
  }
});
Object.defineProperty(exports, "Sticky", {
  enumerable: true,
  get: function () {
    return _Position.Sticky;
  }
});
Object.defineProperty(exports, "Dropdown", {
  enumerable: true,
  get: function () {
    return _Dropdown.default;
  }
});
Object.defineProperty(exports, "FormGroup", {
  enumerable: true,
  get: function () {
    return _FormGroup.default;
  }
});
Object.defineProperty(exports, "InputField", {
  enumerable: true,
  get: function () {
    return _InputField.default;
  }
});
Object.defineProperty(exports, "Label", {
  enumerable: true,
  get: function () {
    return _Label.default;
  }
});
Object.defineProperty(exports, "SelectMenu", {
  enumerable: true,
  get: function () {
    return _SelectMenu.default;
  }
});
Object.defineProperty(exports, "ActionList", {
  enumerable: true,
  get: function () {
    return _ActionList.ActionList;
  }
});
Object.defineProperty(exports, "ActionMenu", {
  enumerable: true,
  get: function () {
    return _ActionMenu.ActionMenu;
  }
});
Object.defineProperty(exports, "DropdownButton", {
  enumerable: true,
  get: function () {
    return _DropdownMenu.DropdownButton;
  }
});
Object.defineProperty(exports, "DropdownMenu", {
  enumerable: true,
  get: function () {
    return _DropdownMenu.DropdownMenu;
  }
});
Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function () {
    return _Button.default;
  }
});
Object.defineProperty(exports, "ButtonDanger", {
  enumerable: true,
  get: function () {
    return _Button.ButtonDanger;
  }
});
Object.defineProperty(exports, "ButtonOutline", {
  enumerable: true,
  get: function () {
    return _Button.ButtonOutline;
  }
});
Object.defineProperty(exports, "ButtonPrimary", {
  enumerable: true,
  get: function () {
    return _Button.ButtonPrimary;
  }
});
Object.defineProperty(exports, "ButtonInvisible", {
  enumerable: true,
  get: function () {
    return _Button.ButtonInvisible;
  }
});
Object.defineProperty(exports, "ButtonTableList", {
  enumerable: true,
  get: function () {
    return _Button.ButtonTableList;
  }
});
Object.defineProperty(exports, "ButtonClose", {
  enumerable: true,
  get: function () {
    return _Button.ButtonClose;
  }
});

var _BorderBox = _interopRequireDefault(require("./BorderBox"));

var _ChoiceFieldset = _interopRequireWildcard(require("./ChoiceFieldset"));

var _ChoiceInputField = _interopRequireDefault(require("./ChoiceInputField"));

var _Flex = _interopRequireDefault(require("./Flex"));

var _Grid = _interopRequireDefault(require("./Grid"));

var _Position = _interopRequireWildcard(require("./Position"));

var _Dropdown = _interopRequireDefault(require("./Dropdown"));

var _FormGroup = _interopRequireDefault(require("./FormGroup"));

var _InputField = _interopRequireDefault(require("./InputField"));

var _Label = _interopRequireDefault(require("./Label"));

var _SelectMenu = _interopRequireDefault(require("./SelectMenu"));

var _ActionList = require("./ActionList");

var _ActionMenu = require("./ActionMenu");

var _DropdownMenu = require("./DropdownMenu");

var _Button = _interopRequireWildcard(require("./Button"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
