// @primer/react/lib/deprecated/ChoiceFieldset/ChoiceFieldsetLegend.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("../..");

var _VisuallyHidden = _interopRequireDefault(require("../../_VisuallyHidden"));

var _ChoiceFieldset = require("./ChoiceFieldset");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ChoiceFieldsetLegend = ({
  children,
  visuallyHidden
}) => /*#__PURE__*/_react.default.createElement(_ChoiceFieldset.Slot, {
  name: "Legend"
}, ({
  required,
  disabled
}) => /*#__PURE__*/_react.default.createElement(_VisuallyHidden.default, {
  as: "legend",
  isVisible: !visuallyHidden,
  title: required ? 'required field' : undefined,
  sx: {
    color: disabled ? 'fg.muted' : undefined,
    fontSize: 2,
    padding: 0
  }
}, required ? /*#__PURE__*/_react.default.createElement(_.Box, {
  display: "flex",
  as: "span"
}, /*#__PURE__*/_react.default.createElement(_.Box, {
  mr: 1
}, children), /*#__PURE__*/_react.default.createElement("span", null, "*")) : children));

ChoiceFieldsetLegend.displayName = "ChoiceFieldsetLegend";
var _default = ChoiceFieldsetLegend;
exports.default = _default;
