// @primer/react/lib-esm/NavList/NavList.stories.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { PageLayout } from '../PageLayout';
import { NavList } from './NavList';
const meta = {
  title: 'Composite components/NavList',
  component: NavList,
  parameters: {
    layout: 'fullscreen'
  }
};
export const Simple = () => /*#__PURE__*/React.createElement(PageLayout, null, /*#__PURE__*/React.createElement(PageLayout.Pane, {
  position: "start"
}, /*#__PURE__*/React.createElement(NavList, null, /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#",
  "aria-current": "page"
}, "Item 1"), /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#"
}, "Item 2"), /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#"
}, "Item 3"))), /*#__PURE__*/React.createElement(PageLayout.Content, null));
Simple.displayName = "Simple";
export const WithSubItems = () => /*#__PURE__*/React.createElement(PageLayout, null, /*#__PURE__*/React.createElement(PageLayout.Pane, {
  position: "start"
}, /*#__PURE__*/React.createElement(NavList, null, /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#"
}, "Item 1"), /*#__PURE__*/React.createElement(NavList.Item, null, "Item 2", /*#__PURE__*/React.createElement(NavList.SubNav, null, /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#",
  "aria-current": "page"
}, "Sub item 1"), /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#"
}, "Sub item 2"))), /*#__PURE__*/React.createElement(NavList.Item, {
  href: "#"
}, "Item 3"))), /*#__PURE__*/React.createElement(PageLayout.Content, null));
WithSubItems.displayName = "WithSubItems";
const ReactRouterLikeLink = /*#__PURE__*/React.forwardRef(({
  to,
  ...props
}, ref) => {
  // eslint-disable-next-line jsx-a11y/anchor-has-content
  return /*#__PURE__*/React.createElement("a", _extends({
    ref: ref,
    href: to
  }, props));
});
export const WithReactRouterLink = () => /*#__PURE__*/React.createElement(PageLayout, null, /*#__PURE__*/React.createElement(PageLayout.Pane, {
  position: "start"
}, /*#__PURE__*/React.createElement(NavList, null, /*#__PURE__*/React.createElement(NavList.Item, {
  as: ReactRouterLikeLink,
  to: "#",
  "aria-current": "page"
}, "Item 1"), /*#__PURE__*/React.createElement(NavList.Item, {
  as: ReactRouterLikeLink,
  to: "#"
}, "Item 2"), /*#__PURE__*/React.createElement(NavList.Item, {
  as: ReactRouterLikeLink,
  to: "#"
}, "Item 3"))), /*#__PURE__*/React.createElement(PageLayout.Content, null));
WithReactRouterLink.displayName = "WithReactRouterLink";
const NextJSLikeLink = /*#__PURE__*/React.forwardRef(({
  href,
  children
}, ref) => {
  const child = React.Children.only(children);
  const childProps = {
    ref,
    href
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.isValidElement(child) ? /*#__PURE__*/React.cloneElement(child, childProps) : null);
});
export const WithNextJSLink = () => /*#__PURE__*/React.createElement(PageLayout, null, /*#__PURE__*/React.createElement(PageLayout.Pane, {
  position: "start"
}, /*#__PURE__*/React.createElement(NavList, null, /*#__PURE__*/React.createElement(NextJSLikeLink, {
  href: "#"
}, /*#__PURE__*/React.createElement(NavList.Item, {
  "aria-current": "page"
}, "Item 1")), /*#__PURE__*/React.createElement(NextJSLikeLink, {
  href: "#"
}, /*#__PURE__*/React.createElement(NavList.Item, null, "Item 2")), /*#__PURE__*/React.createElement(NextJSLikeLink, {
  href: "#"
}, /*#__PURE__*/React.createElement(NavList.Item, null, "Item 3")))), /*#__PURE__*/React.createElement(PageLayout.Content, null));
WithNextJSLink.displayName = "WithNextJSLink";
export default meta;
