// @primer/react/lib/NavList/NavList.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithNextJSLink = exports.WithReactRouterLink = exports.WithSubItems = exports.Simple = void 0;

var _react = _interopRequireDefault(require("react"));

var _PageLayout = require("../PageLayout");

var _NavList = require("./NavList");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const meta = {
  title: 'Composite components/NavList',
  component: _NavList.NavList,
  parameters: {
    layout: 'fullscreen'
  }
};

const Simple = () => /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout, null, /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout.Pane, {
  position: "start"
}, /*#__PURE__*/_react.default.createElement(_NavList.NavList, null, /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  href: "#",
  "aria-current": "page"
}, "Item 1"), /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  href: "#"
}, "Item 2"), /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  href: "#"
}, "Item 3"))), /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout.Content, null));

exports.Simple = Simple;
Simple.displayName = "Simple";

const WithSubItems = () => /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout, null, /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout.Pane, {
  position: "start"
}, /*#__PURE__*/_react.default.createElement(_NavList.NavList, null, /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  href: "#"
}, "Item 1"), /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, null, "Item 2", /*#__PURE__*/_react.default.createElement(_NavList.NavList.SubNav, null, /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  href: "#",
  "aria-current": "page"
}, "Sub item 1"), /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  href: "#"
}, "Sub item 2"))), /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  href: "#"
}, "Item 3"))), /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout.Content, null));

exports.WithSubItems = WithSubItems;
WithSubItems.displayName = "WithSubItems";

const ReactRouterLikeLink = /*#__PURE__*/_react.default.forwardRef(({
  to,
  ...props
}, ref) => {
  // eslint-disable-next-line jsx-a11y/anchor-has-content
  return /*#__PURE__*/_react.default.createElement("a", _extends({
    ref: ref,
    href: to
  }, props));
});

const WithReactRouterLink = () => /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout, null, /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout.Pane, {
  position: "start"
}, /*#__PURE__*/_react.default.createElement(_NavList.NavList, null, /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  as: ReactRouterLikeLink,
  to: "#",
  "aria-current": "page"
}, "Item 1"), /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  as: ReactRouterLikeLink,
  to: "#"
}, "Item 2"), /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  as: ReactRouterLikeLink,
  to: "#"
}, "Item 3"))), /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout.Content, null));

exports.WithReactRouterLink = WithReactRouterLink;
WithReactRouterLink.displayName = "WithReactRouterLink";

const NextJSLikeLink = /*#__PURE__*/_react.default.forwardRef(({
  href,
  children
}, ref) => {
  const child = _react.default.Children.only(children);

  const childProps = {
    ref,
    href
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.isValidElement(child) ? /*#__PURE__*/_react.default.cloneElement(child, childProps) : null);
});

const WithNextJSLink = () => /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout, null, /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout.Pane, {
  position: "start"
}, /*#__PURE__*/_react.default.createElement(_NavList.NavList, null, /*#__PURE__*/_react.default.createElement(NextJSLikeLink, {
  href: "#"
}, /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, {
  "aria-current": "page"
}, "Item 1")), /*#__PURE__*/_react.default.createElement(NextJSLikeLink, {
  href: "#"
}, /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, null, "Item 2")), /*#__PURE__*/_react.default.createElement(NextJSLikeLink, {
  href: "#"
}, /*#__PURE__*/_react.default.createElement(_NavList.NavList.Item, null, "Item 3")))), /*#__PURE__*/_react.default.createElement(_PageLayout.PageLayout.Content, null));

exports.WithNextJSLink = WithNextJSLink;
WithNextJSLink.displayName = "WithNextJSLink";
var _default = meta;
exports.default = _default;
