// @primer/react/lib/_InputValidation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _octiconsReact = require("@primer/octicons-react");

var _react = _interopRequireDefault(require("react"));

var _ = require(".");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const validationIconMap = {
  success: _octiconsReact.CheckCircleFillIcon,
  error: _octiconsReact.AlertFillIcon,
  warning: _octiconsReact.AlertFillIcon
};
const validationColorMap = {
  success: 'success.fg',
  error: 'danger.fg',
  warning: 'attention.fg'
};

const InputValidation = ({
  children,
  id,
  validationStatus,
  sx
}) => {
  const IconComponent = validationStatus ? validationIconMap[validationStatus] : undefined;
  const fgColor = validationStatus ? validationColorMap[validationStatus] : undefined;
  return /*#__PURE__*/_react.default.createElement(_.Text, {
    sx: {
      fontSize: 0,
      fontWeight: 'bold',
      alignItems: 'center',
      color: fgColor,
      display: 'flex',
      a: {
        color: 'currentColor',
        textDecoration: 'underline'
      },
      ...sx
    }
  }, IconComponent && /*#__PURE__*/_react.default.createElement(_.Box, {
    as: "span",
    mr: 1,
    sx: {
      display: 'flex'
    },
    "aria-hidden": "true"
  }, /*#__PURE__*/_react.default.createElement(IconComponent, {
    size: 12,
    fill: "currentColor"
  })), /*#__PURE__*/_react.default.createElement("span", {
    id: id
  }, children));
};

InputValidation.displayName = "InputValidation";
var _default = InputValidation;
exports.default = _default;
