// @primer/react/lib/_InputLabel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require(".");

var _VisuallyHidden = _interopRequireDefault(require("./_VisuallyHidden"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const InputLabel = ({
  children,
  disabled,
  required,
  visuallyHidden,
  htmlFor,
  sx
}) => {
  return /*#__PURE__*/_react.default.createElement(_VisuallyHidden.default, {
    isVisible: !visuallyHidden,
    as: "label",
    htmlFor: htmlFor,
    sx: {
      fontWeight: 'bold',
      fontSize: 1,
      display: 'block',
      color: disabled ? 'fg.muted' : 'fg.default',
      cursor: disabled ? 'default' : 'pointer',
      alignSelf: 'flex-start',
      ...sx
    }
  }, required ? /*#__PURE__*/_react.default.createElement(_.Box, {
    display: "flex",
    as: "span"
  }, /*#__PURE__*/_react.default.createElement(_.Box, {
    mr: 1
  }, children), /*#__PURE__*/_react.default.createElement("span", {
    "aria-hidden": "true"
  }, "*")) : children);
};

InputLabel.displayName = "InputLabel";
var _default = InputLabel;
exports.default = _default;
