// @primer/react/lib/_TextInputInnerVisualSlot.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require(".");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const TextInputInnerVisualSlot = ({
  children,
  hasLoadingIndicator,
  showLoadingIndicator,
  visualPosition
}) => {
  if (!children && !hasLoadingIndicator || visualPosition === 'leading' && !children && !showLoadingIndicator) {
    return null;
  }

  if (!hasLoadingIndicator) {
    return /*#__PURE__*/_react.default.createElement("span", {
      className: "TextInput-icon"
    }, children);
  }

  return /*#__PURE__*/_react.default.createElement("span", {
    className: "TextInput-icon"
  }, /*#__PURE__*/_react.default.createElement(_.Box, {
    display: "flex",
    position: "relative"
  }, children && /*#__PURE__*/_react.default.createElement(_.Box, {
    sx: {
      visibility: showLoadingIndicator ? 'hidden' : 'visible'
    }
  }, children), /*#__PURE__*/_react.default.createElement(_.Spinner, {
    sx: children ? {
      position: 'absolute',
      top: 0,
      height: '100%',
      maxWidth: '100%',
      visibility: showLoadingIndicator ? 'visible' : 'hidden',
      ...(visualPosition === 'leading' ? {
        left: 0
      } : {
        right: 0
      })
    } : {
      visibility: showLoadingIndicator ? 'visible' : 'hidden'
    },
    size: children ? undefined : 'small'
  })));
};

TextInputInnerVisualSlot.displayName = "TextInputInnerVisualSlot";
var _default = TextInputInnerVisualSlot;
exports.default = _default;
