// @primer/react/lib/ActionMenu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionMenu = void 0;

var _react = _interopRequireDefault(require("react"));

var _ssr = require("@react-aria/ssr");

var _octiconsReact = require("@primer/octicons-react");

var _AnchoredOverlay = require("./AnchoredOverlay");

var _hooks = require("./hooks");

var _Divider = require("./ActionList/Divider");

var _ActionListContainerContext = require("./ActionList/ActionListContainerContext");

var _Button = require("./Button");

var _sx = require("./sx");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const MenuContext = /*#__PURE__*/_react.default.createContext({
  renderAnchor: null,
  open: false
});

const Menu = ({
  anchorRef: externalAnchorRef,
  open,
  onOpenChange,
  children
}) => {
  const [combinedOpenState, setCombinedOpenState] = (0, _hooks.useProvidedStateOrCreate)(open, onOpenChange, false);

  const onOpen = _react.default.useCallback(() => setCombinedOpenState(true), [setCombinedOpenState]);

  const onClose = _react.default.useCallback(() => setCombinedOpenState(false), [setCombinedOpenState]);

  const anchorRef = (0, _hooks.useProvidedRefOrCreate)(externalAnchorRef);
  const anchorId = (0, _ssr.useSSRSafeId)();
  let renderAnchor = null; // ðŸš¨ Hack for good API!
  // we strip out Anchor from children and pass it to AnchoredOverlay to render
  // with additional props for accessibility

  const contents = _react.default.Children.map(children, child => {
    if (child.type === MenuButton || child.type === Anchor) {
      renderAnchor = anchorProps => /*#__PURE__*/_react.default.cloneElement(child, anchorProps);

      return null;
    }

    return child;
  });

  return /*#__PURE__*/_react.default.createElement(MenuContext.Provider, {
    value: {
      anchorRef,
      renderAnchor,
      anchorId,
      open: combinedOpenState,
      onOpen,
      onClose
    }
  }, contents);
};

Menu.displayName = "Menu";

const Anchor = /*#__PURE__*/_react.default.forwardRef(({
  children,
  ...anchorProps
}, anchorRef) => {
  return /*#__PURE__*/_react.default.cloneElement(children, { ...anchorProps,
    ref: anchorRef
  });
});
/** this component is syntactical sugar ðŸ­ */


const MenuButton = /*#__PURE__*/_react.default.forwardRef(({
  sx: sxProp = {},
  ...props
}, anchorRef) => {
  return /*#__PURE__*/_react.default.createElement(Anchor, {
    ref: anchorRef
  }, /*#__PURE__*/_react.default.createElement(_Button.Button, _extends({
    type: "button",
    trailingIcon: _octiconsReact.TriangleDownIcon,
    sx: (0, _sx.merge)({
      // override the margin on caret for optical alignment
      '[data-component=trailingIcon]': {
        marginX: -1
      }
    }, sxProp)
  }, props)));
});

const Overlay = ({
  children,
  align = 'start',
  ...overlayProps
}) => {
  // we typecast anchorRef as required instead of optional
  // because we know that we're setting it in context in Menu
  const {
    anchorRef,
    renderAnchor,
    anchorId,
    open,
    onOpen,
    onClose
  } = _react.default.useContext(MenuContext);

  const containerRef = /*#__PURE__*/_react.default.createRef();

  const {
    openWithFocus
  } = (0, _hooks.useMenuInitialFocus)(open, onOpen, containerRef);
  (0, _hooks.useMnemonics)(open, containerRef);
  return /*#__PURE__*/_react.default.createElement(_AnchoredOverlay.AnchoredOverlay, {
    anchorRef: anchorRef,
    renderAnchor: renderAnchor,
    anchorId: anchorId,
    open: open,
    onOpen: openWithFocus,
    onClose: onClose,
    align: align,
    overlayProps: overlayProps,
    focusZoneSettings: {
      focusOutBehavior: 'wrap'
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    ref: containerRef
  }, /*#__PURE__*/_react.default.createElement(_ActionListContainerContext.ActionListContainerContext.Provider, {
    value: {
      container: 'ActionMenu',
      listRole: 'menu',
      listLabelledBy: anchorId,
      selectionAttribute: 'aria-checked',
      // Should this be here?
      afterSelect: onClose
    }
  }, children)));
};

Overlay.displayName = "Overlay";
Menu.displayName = 'ActionMenu';
const ActionMenu = Object.assign(Menu, {
  Button: MenuButton,
  Anchor,
  Overlay,
  Divider: _Divider.Divider
});
exports.ActionMenu = ActionMenu;
