// @primer/react/lib/Label.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.variants = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _sx = _interopRequireDefault(require("./sx"));

var _constants = require("./constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const variants = {
  default: {
    borderColor: 'border.default'
  },
  primary: {
    borderColor: 'fg.default'
  },
  secondary: {
    borderColor: 'border.muted',
    color: 'fg.muted'
  },
  accent: {
    borderColor: 'accent.emphasis',
    color: 'accent.fg'
  },
  success: {
    borderColor: 'success.emphasis',
    color: 'success.fg'
  },
  attention: {
    borderColor: 'attention.emphasis',
    color: 'attention.fg'
  },
  severe: {
    borderColor: 'severe.emphasis',
    color: 'severe.fg'
  },
  danger: {
    borderColor: 'danger.emphasis',
    color: 'danger.fg'
  },
  done: {
    borderColor: 'done.fg',
    color: 'done.emphasis'
  },
  sponsors: {
    borderColor: 'sponsors.fg',
    color: 'sponsors.emphasis'
  }
};
exports.variants = variants;
const sizes = {
  small: {
    height: '20px',
    padding: '0 7px' // hard-coded to align with Primer ViewComponents and Primer CSS

  },
  large: {
    height: '24px',
    padding: '0 10px' // hard-coded to align with Primer ViewComponents and Primer CSS

  }
};

const Label = _styledComponents.default.span.withConfig({
  displayName: "Label",
  componentId: "sc-6dyj7v-0"
})(["align-items:center;background-color:transparent;border-width:1px;border-radius:999px;border-style:solid;display:inline-flex;font-weight:", ";font-size:", ";line-height:1;white-space:nowrap;", ";", ";", ";"], (0, _constants.get)('fontWeights.bold'), (0, _constants.get)('fontSizes.0'), (0, _styledSystem.variant)({
  variants
}), (0, _styledSystem.variant)({
  prop: 'size',
  variants: sizes
}), _sx.default);

Label.defaultProps = {
  size: 'small',
  variant: 'default'
};
var _default = Label;
exports.default = _default;
