// @primer/react/lib/PointerBox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Box = _interopRequireDefault(require("./Box"));

var _Caret = _interopRequireDefault(require("./Caret"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function PointerBox(props) {
  // don't destructure these, just grab them
  const {
    bg,
    border,
    borderColor,
    theme,
    sx
  } = props;
  const {
    caret,
    children,
    ...boxProps
  } = props;
  const caretProps = {
    bg: bg || (sx === null || sx === void 0 ? void 0 : sx.bg) || (sx === null || sx === void 0 ? void 0 : sx.backgroundColor),
    borderColor: borderColor || (sx === null || sx === void 0 ? void 0 : sx.borderColor),
    borderWidth: border,
    location: caret,
    theme
  };
  const defaultBoxProps = {
    borderWidth: '1px',
    borderStyle: 'solid',
    borderColor: 'border.default',
    borderRadius: 2
  };
  return /*#__PURE__*/_react.default.createElement(_Box.default, _extends({}, defaultBoxProps, boxProps, {
    sx: { ...sx,
      position: 'relative'
    }
  }), children, /*#__PURE__*/_react.default.createElement(_Caret.default, caretProps));
}

PointerBox.displayName = "PointerBox";
var _default = PointerBox;
exports.default = _default;
