// @primer/react/lib/ProgressBar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _constants = require("./constants");

var _sx = _interopRequireDefault(require("./sx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Bar = _styledComponents.default.span.withConfig({
  displayName: "ProgressBar__Bar",
  componentId: "sc-bfftmc-0"
})(["width:", ";", ";"], props => props.progress ? `${props.progress}%` : 0, _sx.default);

const sizeMap = {
  small: '5px',
  large: '10px',
  default: '8px'
};

const ProgressContainer = _styledComponents.default.span.withConfig({
  displayName: "ProgressBar__ProgressContainer",
  componentId: "sc-bfftmc-1"
})(["display:", ";overflow:hidden;background-color:", ";border-radius:", ";height:", ";", " ", ";"], props => props.inline ? 'inline-flex' : 'flex', (0, _constants.get)('colors.border.default'), (0, _constants.get)('radii.1'), props => sizeMap[props.barSize || 'default'], _styledSystem.width, _sx.default);

function ProgressBar({
  progress,
  bg,
  ...rest
}) {
  return /*#__PURE__*/_react.default.createElement(ProgressContainer, rest, /*#__PURE__*/_react.default.createElement(Bar, {
    progress: progress,
    sx: {
      bg
    }
  }));
}

ProgressBar.displayName = "ProgressBar";
ProgressBar.defaultProps = {
  bg: 'success.emphasis',
  barSize: 'default'
};
var _default = ProgressBar;
exports.default = _default;
