// @primer/react/lib/TabNav.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classnames = _interopRequireDefault(require("classnames"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _constants = require("./constants");

var _useFocusZone = require("./hooks/useFocusZone");

var _sx = _interopRequireDefault(require("./sx"));

var _getGlobalFocusStyles = _interopRequireDefault(require("./_getGlobalFocusStyles"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const ITEM_CLASS = 'TabNav-item';
const SELECTED_CLASS = 'selected';

const TabNavBase = _styledComponents.default.div.withConfig({
  displayName: "TabNav__TabNavBase",
  componentId: "sc-1ab407u-0"
})(["", ""], _sx.default);

const TabNavTabList = _styledComponents.default.div.withConfig({
  displayName: "TabNav__TabNavTabList",
  componentId: "sc-1ab407u-1"
})(["display:flex;margin-bottom:-1px;overflow:auto;"]);

const TabNavNav = _styledComponents.default.nav.withConfig({
  displayName: "TabNav__TabNavNav",
  componentId: "sc-1ab407u-2"
})(["margin-top:0;border-bottom:1px solid ", ";"], (0, _constants.get)('colors.border.default'));

function TabNav({
  children,
  'aria-label': ariaLabel,
  ...rest
}) {
  const customContainerRef = (0, _react.useRef)(null);

  const customStrategy = _react.default.useCallback(() => {
    if (customContainerRef.current) {
      const tabs = Array.from(customContainerRef.current.querySelectorAll('[role=tab][aria-selected=true]'));
      return tabs[0];
    }
  }, [customContainerRef]);

  const {
    containerRef: navRef
  } = (0, _useFocusZone.useFocusZone)({
    containerRef: customContainerRef,
    bindKeys: _useFocusZone.FocusKeys.ArrowHorizontal | _useFocusZone.FocusKeys.HomeAndEnd,
    focusOutBehavior: 'wrap',
    focusInStrategy: customStrategy
  });
  return /*#__PURE__*/_react.default.createElement(TabNavBase, _extends({}, rest, {
    ref: navRef
  }), /*#__PURE__*/_react.default.createElement(TabNavNav, {
    "aria-label": ariaLabel
  }, /*#__PURE__*/_react.default.createElement(TabNavTabList, {
    role: "tablist"
  }, children)));
}

TabNav.displayName = "TabNav";

const TabNavLink = _styledComponents.default.a.attrs(props => ({
  activeClassName: typeof props.to === 'string' ? 'selected' : '',
  className: (0, _classnames.default)(ITEM_CLASS, props.selected && SELECTED_CLASS, props.className),
  role: 'tab',
  'aria-selected': !!props.selected,
  tabIndex: -1
})).withConfig({
  displayName: "TabNav__TabNavLink",
  componentId: "sc-1ab407u-3"
})(["padding:8px 12px;font-size:", ";line-height:20px;color:", ";text-decoration:none;background-color:transparent;border:1px solid transparent;border-bottom:0;", ";&:hover,&:focus{color:", ";text-decoration:none;}&.selected{color:", ";border-color:", ";border-top-right-radius:", ";border-top-left-radius:", ";background-color:", ";}", ";"], (0, _constants.get)('fontSizes.1'), (0, _constants.get)('colors.fg.default'), (0, _getGlobalFocusStyles.default)('-6px'), (0, _constants.get)('colors.fg.default'), (0, _constants.get)('colors.fg.default'), (0, _constants.get)('colors.border.default'), (0, _constants.get)('radii.2'), (0, _constants.get)('radii.2'), (0, _constants.get)('colors.canvas.default'), _sx.default);

TabNavLink.displayName = 'TabNav.Link';

var _default = Object.assign(TabNav, {
  Link: TabNavLink
});

exports.default = _default;
