// @primer/react/lib/_CheckboxOrRadioGroup/_CheckboxOrRadioGroupLabel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("..");

var _VisuallyHidden = _interopRequireDefault(require("../_VisuallyHidden"));

var _slots = require("./slots");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const CheckboxOrRadioGroupLabel = ({
  children,
  visuallyHidden,
  sx
}) => /*#__PURE__*/_react.default.createElement(_slots.Slot, {
  name: "Label"
}, ({
  required,
  disabled
}) => /*#__PURE__*/_react.default.createElement(_VisuallyHidden.default, {
  isVisible: !visuallyHidden,
  title: required ? 'required field' : undefined,
  sx: {
    display: 'block',
    color: disabled ? 'fg.muted' : undefined,
    fontSize: 2,
    ...sx
  }
}, required ? /*#__PURE__*/_react.default.createElement(_.Box, {
  display: "flex",
  as: "span"
}, /*#__PURE__*/_react.default.createElement(_.Box, {
  mr: 1
}, children), /*#__PURE__*/_react.default.createElement("span", null, "*")) : children));

CheckboxOrRadioGroupLabel.displayName = "CheckboxOrRadioGroupLabel";
CheckboxOrRadioGroupLabel.defaultProps = {
  visuallyHidden: false
};
var _default = CheckboxOrRadioGroupLabel;
exports.default = _default;
