// @primer/react/lib/ActionList/Divider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Divider = void 0;

var _react = _interopRequireDefault(require("react"));

var _Box = _interopRequireDefault(require("../Box"));

var _constants = require("../constants");

var _sx = require("../sx");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Visually separates `Item`s or `Group`s in an `ActionList`.
 */
const Divider = ({
  sx = {}
}) => {
  return /*#__PURE__*/_react.default.createElement(_Box.default, {
    as: "li",
    "aria-hidden": "true",
    sx: (0, _sx.merge)({
      height: 1,
      backgroundColor: 'actionListItem.inlineDivider',
      marginTop: theme => `calc(${(0, _constants.get)('space.2')(theme)} - 1px)`,
      marginBottom: 2,
      listStyle: 'none' // hide the ::marker inserted by browser's stylesheet

    }, sx),
    "data-component": "ActionList.Divider"
  });
};

exports.Divider = Divider;
Divider.displayName = "Divider";
