// @primer/react/lib/ActionList/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionList = void 0;

var _List = require("./List");

var _Group = require("./Group");

var _Item = require("./Item");

var _LinkItem = require("./LinkItem");

var _Divider = require("./Divider");

var _Description = require("./Description");

var _Visuals = require("./Visuals");

/**
 * Collection of list-related components.
 */
const ActionList = Object.assign(_List.List, {
  /** Collects related `Items` in an `ActionList`. */
  Group: _Group.Group,

  /** An actionable or selectable `Item` */
  Item: _Item.Item,

  /** A `Item` that renders a full-size anchor inside ListItem */
  LinkItem: _LinkItem.LinkItem,

  /** Visually separates `Item`s or `Group`s in an `ActionList`. */
  Divider: _Divider.Divider,

  /** Secondary text which provides additional information about an `Item`. */
  Description: _Description.Description,

  /** Icon (or similar) positioned before `Item` text. */
  LeadingVisual: _Visuals.LeadingVisual,

  /** Icon (or similar) positioned after `Item` text. */
  TrailingVisual: _Visuals.TrailingVisual
});
exports.ActionList = ActionList;
