// @primer/react/lib/FilteredActionList/FilteredActionList.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = Default;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("..");

var _FilteredActionList = require("../FilteredActionList");

var _BaseStyles = _interopRequireDefault(require("../BaseStyles"));

var _Box = _interopRequireDefault(require("../Box"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const meta = {
  title: 'Composite components/FilteredActionList',
  component: _FilteredActionList.FilteredActionList,
  decorators: [Story => /*#__PURE__*/_react.default.createElement(_.ThemeProvider, null, /*#__PURE__*/_react.default.createElement(_BaseStyles.default, null, /*#__PURE__*/_react.default.createElement(Story, null)))],
  parameters: {
    controls: {
      disable: true
    }
  }
};
var _default = meta;
exports.default = _default;

function getColorCircle(color) {
  return function () {
    return /*#__PURE__*/_react.default.createElement(_Box.default, {
      bg: color,
      borderColor: color,
      width: 14,
      height: 14,
      borderRadius: 10,
      margin: "auto",
      borderWidth: "1px",
      borderStyle: "solid"
    });
  };
}

const items = [{
  leadingVisual: getColorCircle('#a2eeef'),
  text: 'enhancement',
  id: 1
}, {
  leadingVisual: getColorCircle('#d73a4a'),
  text: 'bug',
  id: 2
}, {
  leadingVisual: getColorCircle('#0cf478'),
  text: 'good first issue',
  id: 3
}, {
  leadingVisual: getColorCircle('#ffd78e'),
  text: 'design',
  id: 4
}, {
  leadingVisual: getColorCircle('#ff0000'),
  text: 'blocker',
  id: 5
}, {
  leadingVisual: getColorCircle('#a4f287'),
  text: 'backend',
  id: 6
}, {
  leadingVisual: getColorCircle('#8dc6fc'),
  text: 'frontend',
  id: 7
}];

function Default() {
  const [filter, setFilter] = _react.default.useState('');

  const filteredItems = items.filter(item => item.text.toLowerCase().startsWith(filter.toLowerCase()));
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("h1", null, "Filtered Action List"), /*#__PURE__*/_react.default.createElement("div", null, "Please select labels that describe your issue:"), /*#__PURE__*/_react.default.createElement(_FilteredActionList.FilteredActionList, {
    placeholderText: "Filter Labels",
    items: filteredItems,
    onFilterChange: setFilter,
    sx: {
      border: '1px solid',
      padding: '8px'
    }
  }));
}
